src/main/java/ui/MenuControl.java

```diff
@@ -129,12 +131,31 @@ public void changed(ObservableValue<? extends Number> arg0, Number arg1, Number
         cols.getItems().addAll(createRight, createLeft, closePanel);
         return cols;
     }
+    
+    private void onBoardSave() {
+        logger.info("Menu: Boards > Save");
+        
+        List<PanelInfo> panels = getCurrentPanels();
+        
+        if (openBoardName.equals("")) {
+            onBoardSaveAs();
+        }
```

Comment:
Also, if there should be a return, you could swap the conditional and the `getCurrentPanels()` line, to make it clear that the absence of a board name immediately delegates to `onBoardSaveAs` without needing to do anything else.